common/src/main/kotlin/DiscordTimestamp.kt

```diff
@@ -0,0 +1,87 @@
+package dev.kord.common
+
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toInstant
+import kotlinx.serialization.Serializable
+
+/**
+ * The [timestamp format](https://discord.com/developers/docs/reference#message-formatting-formats) of the message formats
+ */
+@Serializable
+data class DiscordTimestamp(
+
+    /**
+     * The Unix timestamp to be displayed
+     */
+    val time: Long,
+
+    /**
+     * The style of the timestamp
+     */
+    val style: DiscordTimestampStyle = DiscordTimestampStyle.ShortDateTime
+) : Comparable<DiscordTimestamp> {
+
+    val value: String
+        get() = "<t:$time:${style.style}>"
+
+    override fun equals(other: Any?): Boolean = other is DiscordTimestamp && other.time == time
+    operator fun plus(value: Long): DiscordTimestamp = DiscordTimestamp(time + value, style)
+    operator fun minus(value: Long): DiscordTimestamp = DiscordTimestamp(time - value, style)
+
+    override fun compareTo(other: DiscordTimestamp): Int = when {
+        time > other.time -> 1
+        time < other.time -> -1
+        else -> 0
+    }
+
+    override fun toString() = value
+}
+
+fun LocalDateTime.toDiscordTimestamp(style: DiscordTimestampStyle? = null) =
+    DiscordTimestamp(toInstant(TimeZone.UTC).toEpochMilliseconds(), style ?: DiscordTimestampStyle.ShortDateTime)
+
+fun Instant.toDiscordTimestamp(style: DiscordTimestampStyle? = null) =
+    DiscordTimestamp(toEpochMilliseconds(), style ?: DiscordTimestampStyle.ShortDateTime)
```

Comment:
I would make `LocalDateTime.toDiscordTimestamp` call `Instant.toDiscordTimestamp` but this is just a nit-pick.
And use `DiscordTimestampStyle.ShorrDateTime` as the default for `style`, no need for this to be nullable.